core/controllers/library.py

```diff
@@ -143,6 +156,87 @@ def get(self):
         self.render_json(self.values)
 
 
+class LibraryGroupPage(base.BaseHandler):
+    """The page for displaying top rated and recently published explorations
+    """
+
+    def get(self, rank_method):
+        """Handles GET requests."""
+
+        search_mode = 'search' in self.request.url
+        rank_mode = 'explorations' in self.request.url
+
+        if rank_method == 'recently_published':
+            title = feconf.LIBRARY_CATEGORY_RECENTLY_PUBLISHED
+        elif rank_method == 'top_rated':
+            title = feconf.LIBRARY_CATEGORY_TOP_RATED_EXPLORATIONS
+        else:
+            raise self.PageNotFoundException
+
+        self.values.update({
+            'meta_description': (
+                feconf.SEARCH_PAGE_DESCRIPTION if search_mode
+                else feconf.LIBRARY_PAGE_DESCRIPTION),
+            'nav_mode': feconf.NAV_MODE_LIBRARY,
+            'has_fully_registered': bool(
+                self.user_id and
+                user_services.has_fully_registered(self.user_id)),
+            'LANGUAGE_CODES_AND_NAMES': (
+                utils.get_all_language_codes_and_names()),
+            'search_mode': search_mode,
+            'rank_mode': rank_mode,
+            'title': title,
+            'SEARCH_DROPDOWN_CATEGORIES': feconf.SEARCH_DROPDOWN_CATEGORIES,
+        })
+        self.render_template('library/library.html')
+
+
+class LibraryGroupIndexHandler(base.BaseHandler):
+    """Provides data for categories such as top rated and recently published"""
+
+    def get(self, rank_method):
+        summary_dicts_by_category = []
+
+        if rank_method == 'recently_published':
+            recently_published_summary_dicts = (
+                summary_services.get_recently_published_exploration_summary_dicts( # pylint: disable=line-too-long
+                    feconf.RECENTLY_PUBLISHED_QUERY_LIMIT_FULL_PAGE))
+            if recently_published_summary_dicts:
+                summary_dicts_by_category.insert(0, {
```

Comment:
You could just append or say `= [...]`? I mean, there's at most one, and the variable is an empty list before that.

Ditto below.